File: cosmoauthorization/src/java/it/csi/cosmo/cosmoauthorization/business/service/ConfigurazioneEnteService.java
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */
package it.csi.cosmo.cosmoauthorization.business.service;

import java.math.BigDecimal;
import java.util.List;
import it.csi.cosmo.cosmoauthorization.dto.rest.ConfigurazioneEnte;

/**
 * Servizio per la gestione della configurazione dell'ente
 */
public interface ConfigurazioneEnteService {

  ConfigurazioneEnte getConfigurazioneEnte(BigDecimal idEnte, String chiave);

  List<ConfigurazioneEnte> getConfigurazioniEnte(BigDecimal idEnte);

  ConfigurazioneEnte postConfigurazioneEnte(BigDecimal idEnte, ConfigurazioneEnte configurazione);

  ConfigurazioneEnte putConfigurazioneEnte(BigDecimal idEnte, String chiave,
      ConfigurazioneEnte configurazione);

  void deleteConfigurazioneEnte(BigDecimal idEnte, String chiave);

}

File: cosmowcl/src/app/shared/services/configurazione-ente.service.ts
/*
 * Copyright CSI-Piemonte - 2023
 * SPDX-License-Identifier: GPL-3.0-or-later
 */
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ConfigurazioneEnte } from '../models/api/cosmoauthorization/configurazioneEnte';
import { ApiUrls } from '../utilities/apiurls';
import { Utils } from '../utilities/utilities';

@Injectable()
export class ConfigurazioneEnteService {

  constructor(private http: HttpClient) {
    // NOP
  }

  getConfigurazioneEnte(chiave: string, idEnte?: number): Observable<ConfigurazioneEnte> {
    Utils.require(chiave);

    const options = idEnte ?  { params: new HttpParams().set('idEnte', idEnte.toString()) } : {};
    return this.http.get<ConfigurazioneEnte>(ApiUrls.CONFIGURAZIONE_ENTE_CON_CHIAVE.replace('{chiave}', chiave), options);
  }

  getConfigurazioniEnte(idEnte?: number): Observable<ConfigurazioneEnte[]> {
    const options = idEnte ?  { params: new HttpParams().set('idEnte', idEnte.toString()) } : {};
    return this.http.get<ConfigurazioneEnte[]>(ApiUrls.CONFIGURAZIONE_ENTE, options);
  }

  deleteConfigurazioneEnte(chiave: string, idEnte?: number): Observable<void> {
    Utils.require(chiave);
    const options = idEnte ?  { params: new HttpParams().set('idEnte', idEnte.toString()) } : {};
    return this.http.delete<void>(ApiUrls.CONFIGURAZIONE_ENTE_CON_CHIAVE.replace('{chiave}', chiave), options);
  }

  creaConfigurazioneEnte(request: ConfigurazioneEnte, idEnte?: number): Observable<ConfigurazioneEnte> {
    Utils.require(request);
    const options = idEnte ?  { params: new HttpParams().set('idEnte', idEnte.toString()) } : {};
    return this.http.post<ConfigurazioneEnte>(ApiUrls.CONFIGURAZIONE_ENTE, request, options);
  }

  aggiornaConfigurazioneEnte(chiave: string, request: ConfigurazioneEnte, idEnte?: number): Observable<ConfigurazioneEnte> {
    Utils.require(chiave);
    Utils.require(request);
    const options = idEnte ?  { params: new HttpParams().set('idEnte', idEnte.toString()) } : {};

    return this.http.put<ConfigurazioneEnte>(ApiUrls.CONFIGURAZIONE_ENTE_CON_CHIAVE.replace('{chiave}', chiave), request, options);
  }
}
